### ISSUE TITLE
F.dropout crashes with LongTensor

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#25245]
It seems `F.dropout` crashes with the input tensor has dtype LongTensor instead of raising an Exception

e.g.
```python
>>> a = torch.tensor([1, 2, 3, 4])  # instead of torch.FloatTensor(...)
>>> F.dropout(a, 0.5)
Floating point exception (core dumped)
```

```
 - PyTorch Version: 1.3.0.dev20190826
 - OS: Ubuntu 18.04
 - How you installed PyTorch: conda
 - Python version: 3.7
 - CUDA/cuDNN version: 10.0
```

**Comment by andreaskoepf** [comment]
This bug can be reproduced (without dropout) by dividing a LongTensor by a floating scalar with absolute value `< 1.0`:

```
x = torch.tensor([1,2,3])
x / 0.5
```
-> Floating point exception (core dumped)

**Comment by andreaskoepf** [comment]
With a cuda tensor this does not crash but has also strange results:

```
>>> c = torch.tensor([1,2,3]).cuda()
>>> c / 2.5
tensor([0, 1, 1], device='cuda:0')
>>> c / 0.5
tensor([4294967295, 4294967295, 4294967295], device='cuda:0')
>>> c / 0.1
tensor([4294967295, 4294967295, 4294967295], device='cuda:0')
>>> c / -0.0001
tensor([4294967295, 4294967295, 4294967295], device='cuda:0')
```
